### PR TITLE
Add new omrsysinfo_get_CPU_load API

### PR DESCRIPTION
**Pull request by fjeremic**
Makes use of eclipse/omr#4702 to implement the `getSystemCpuLoad` OperatingSystemMXBean. The existing logic (removed code in this PR) was effectively sunk down into OMR and z/OS support was subsequently added. There should be no change in behavior for existing platforms, and the test coverage proves this is the case.

### CODE REVIEW

**Comment by fjeremic** [comment]
This PR has a soft dependency on the OMR PR, meaning they are not cross-dependent, however this PR should not be merged until the OMR PR has propagated.

**Comment by fjeremic** [comment]
Jenkins test all sanity jdk8 depends eclipse/omr#4702

**Comment by fjeremic** [comment]
Jenkins test xlinux sanity jdk8 depends eclipse/omr#4702

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Comment by fjeremic** [comment]
@rwy0717 this one should be ready as well. This is the OpenJ9 portion of the work making use of the OMR API we merged. The OMR change has propagated.

File: runtime/oti/j9port_generated.h [hunk 1]
```diff
@@ -620,6 +620,7 @@ extern J9_CFUNC int32_t j9port_isCompatible(struct J9PortLibraryVersion *expecte
 #define j9sysinfo_get_groupname(param1,param2) OMRPORT_FROM_J9PORT(privatePortLibrary)->sysinfo_get_groupname(OMRPORT_FROM_J9PORT(privatePortLibrary),param1,param2)
 #define j9sysinfo_get_load_average(param1) OMRPORT_FROM_J9PORT(privatePortLibrary)->sysinfo_get_load_average(OMRPORT_FROM_J9PORT(privatePortLibrary),param1)
 #define j9sysinfo_get_CPU_utilization(param1) OMRPORT_FROM_J9PORT(privatePortLibrary)->sysinfo_get_CPU_utilization(OMRPORT_FROM_J9PORT(privatePortLibrary),param1)
+#define j9sysinfo_get_CPU_load(param1) OMRPORT_FROM_J9PORT(privatePortLibrary)->sysinfo_get_CPU_load(OMRPORT_FROM_J9PORT(privatePortLibrary),param1)
```
Comment (pshipton) [inline runtime/oti/j9port_generated.h]: I don't see why this is needed, the code could call omrsysinfo directly. I assume the existing macros were added to avoid having to change all the OpenJ9 to to call omr. Perhaps subsequent changes have followed the same pattern for no reason, but it would be clearer to call omr directly for anything new.

**Comment by pshipton** [comment]
Original PR testing (all passed) https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/2760/

**Comment by pshipton** [comment]
Note the test runs in extended testing, not sanity.

File: runtime/oti/j9port_generated.h (hunk 1 above)
Comment (fjeremic) [reply runtime/oti/j9port_generated.h]: I agree. I wondered that myself. Fixed in 1f3ba30

File: runtime/jcl/common/mgmtosext.c [hunk 2]
```diff
@@ -79,6 +79,32 @@ Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getProce
 	return (0 == rc)? (jlong) size: (jlong) -1;
 }
 
+jdouble JNICALL
+Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getSystemCpuLoadImpl(JNIEnv *env, jobject instance) {
+	PORT_ACCESS_FROM_ENV(env);
+	double cpuLoad;
```
[the remark this answers is not in the document]
Comment (fjeremic) [reply runtime/jcl/common/mgmtosext.c]: Fixed in 1f3ba30

**Comment by rwy0717** [comment]
I think, the intention of the wrapper macros is to allow the j9 port library to function as a clean superset of the omr port library. Although, I agree that calling straight to omr is a bit clearer.

Without the wrapper, you need to use [`OMRPORT_ACCESS_FROM_j9PORT`](https://github.com/eclipse/openj9/blob/master/runtime/oti/j9port.h#L64):

```c++
J9PORT_ACCESS_FROM_ENV(env);
OMRPORT_ACCESS_FROM_j9PORT(PORTLIB);
```

File: runtime/jcl/common/mgmtosext.c [hunk 3]
```diff
@@ -79,6 +79,33 @@ Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getProce
 	return (0 == rc)? (jlong) size: (jlong) -1;
 }
 
+jdouble JNICALL
+Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getSystemCpuLoadImpl(JNIEnv *env, jobject instance) {
+	PORT_ACCESS_FROM_ENV(env);
+   OMRPORT_ACCESS_FROM_J9PORT(portLibrary);
```
Comment (rwy0717) [inline runtime/jcl/common/mgmtosext.c]: should be [`PORTLIB`](https://github.com/eclipse/openj9/blob/master/runtime/oti/j9port.h#L56), I think `portLibrary` won't be defined.

Comment (fjeremic) [reply runtime/jcl/common/mgmtosext.c]: Fixed in e200fcd

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java [hunk 4]
```diff
@@ -426,9 +424,11 @@ public final long getProcessVirtualMemorySize() {
 	 */
 	@Override
 	public final double getSystemCpuLoad() {
-		return cpuUtilizationHelper.getSystemCpuLoad();
+		return this.getSystemCpuLoadImpl();
 	}
 
+	private native double getSystemCpuLoadImpl();
```
Comment (keithc-ca) [inline jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java]: This method can be `static`.

File: runtime/jcl/common/mgmtosext.c [hunk 5]
```diff
@@ -79,6 +79,33 @@ Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getProce
 	return (0 == rc)? (jlong) size: (jlong) -1;
 }
 
+jdouble JNICALL
+Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getSystemCpuLoadImpl(JNIEnv *env, jobject instance) {
+	PORT_ACCESS_FROM_ENV(env);
+   OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
```
Comment (keithc-ca) [inline runtime/jcl/common/mgmtosext.c]: Inconsistent indentation.

File: runtime/oti/j9port_generated.h [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 1991, 2019 IBM Corp. and others
+ * Copyright (c) 1991, 2020 IBM Corp. and others
```
Comment (keithc-ca) [inline runtime/oti/j9port_generated.h]: Please undo this change - the file is otherwise unmodified.

File: runtime/oti/jclprots.h [hunk 7]
```diff
@@ -140,8 +140,8 @@ extern J9_CFUNC jlong JNICALL
 Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getProcessPrivateMemorySizeImpl (JNIEnv *env, jobject instance);
 extern J9_CFUNC jlong JNICALL
 Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getProcessPhysicalMemorySizeImpl (JNIEnv *env, jobject instance);
-extern J9_CFUNC jobject JNICALL
-Java_com_ibm_lang_management_internal_SysinfoCpuTime_getCpuUtilizationImpl (JNIEnv *env, jclass clazz);
+extern J9_CFUNC jdouble JNICALL
+Java_com_ibm_lang_management_internal_ExtendedOperatingSystemMXBeanImpl_getSystemCpuLoadImpl (JNIEnv *env, jobject instance);
```
Comment (keithc-ca) [inline runtime/oti/jclprots.h]: The signature will need to be updated if made `static`.

File: test/functional/JLM_Tests/src/org/openj9/test/management/TestGetSystemCpuLoad.java [hunk 8]
```diff
@@ -38,97 +38,63 @@
...
-			delayMillis(1000);
-			load = ibmBean1.getSystemCpuLoad();
-			validateLoad(load, false, NO_ERROR, "getSystemCpuLoad after 1 second");
+			testMxBeanImpl((com.ibm.lang.management.OperatingSystemMXBean)osBean);
 		} else {
 			Assert.fail("OperatingSystemMXBean is wrong type: " + osBean.getClass().getName());
 		}
 	}
 
+	public static void testMxBeanImpl(com.ibm.lang.management.OperatingSystemMXBean ibmBean) {
```
Comment (keithc-ca) [inline test/functional/JLM_Tests/src/org/openj9/test/management/TestGetSystemCpuLoad.java]: Why is this `public` and not `private`?

File: runtime/oti/j9port_generated.h (hunk 6 above)
Comment (fjeremic) [reply runtime/oti/j9port_generated.h]: Fixed in [force-push](https://github.com/eclipse/openj9/compare/e200fcd0089727bb8c11ffa2d51bc48880865951..a15fd0a5f1583c638085002e198a4ef9732245cf).

File: runtime/jcl/common/mgmtosext.c (hunk 5 above)
Comment (fjeremic) [reply runtime/jcl/common/mgmtosext.c]: Fixed in [force-push](https://github.com/eclipse/openj9/compare/e200fcd0089727bb8c11ffa2d51bc48880865951..a15fd0a5f1583c638085002e198a4ef9732245cf).

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java (hunk 4 above)
Comment (fjeremic) [reply jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java]: Fixed in [force-push](https://github.com/eclipse/openj9/compare/e200fcd0089727bb8c11ffa2d51bc48880865951..a15fd0a5f1583c638085002e198a4ef9732245cf).

File: test/functional/JLM_Tests/src/org/openj9/test/management/TestGetSystemCpuLoad.java (hunk 8 above)
Comment (fjeremic) [reply test/functional/JLM_Tests/src/org/openj9/test/management/TestGetSystemCpuLoad.java]: Fixed in [force-push](https://github.com/eclipse/openj9/compare/e200fcd0089727bb8c11ffa2d51bc48880865951..a15fd0a5f1583c638085002e198a4ef9732245cf).

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java (hunk 4 above)
Comment (fjeremic) [reply jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java]: The reason this was not done was to remain consistent with the other APIs within this same class. For example, by the same logic the following APIs could be made `static` as well:

https://github.com/eclipse/openj9/blob/0ae4f22e2d7769fb2796382ef2b33bcacb749ff2/jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java#L338

https://github.com/eclipse/openj9/blob/0ae4f22e2d7769fb2796382ef2b33bcacb749ff2/jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java#L353

https://github.com/eclipse/openj9/blob/0ae4f22e2d7769fb2796382ef2b33bcacb749ff2/jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java#L391

https://github.com/eclipse/openj9/blob/0ae4f22e2d7769fb2796382ef2b33bcacb749ff2/jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java#L422

etc. (there are dozens more). As such making this new API `static` would be inconsistent with the others. Would it be reasonable to not deviate from these other similar APIs, and instead open up an issue to bulk-change them all to be `static`?

Comment (keithc-ca) [reply jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java]: I'm fine with following the pattern and opening an issue to clean up later (unless a static native is more expensive than a non-static one).

**Comment by pshipton** [comment]
jenkins test extended plinux,win32,aix,osx jdk8

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java (hunk 4 above)
Comment (fjeremic) [reply jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedOperatingSystemMXBeanImpl.java]: I've opened up #8647 to track this work. Thanks for identifying the opportunity.
